site/tests/integration/blockpage.test.ts

```diff
@@ -0,0 +1,128 @@
+import { expect, test } from "playwright-test-coverage";
+
+import blocksData from "../../blocks-data.json";
+import type { ExpandedBlockMetadata } from "../../src/lib/blocks";
+
+const typedBlocksData = blocksData as ExpandedBlockMetadata[];
+
+const codeBlock = typedBlocksData.find(
+  ({ name }) => name === "@hashintel/block-code",
+);
+
+const unsupportedBlock = typedBlocksData.find(
+  ({ name }) => name === "@hashintel/block-embed",
+);
+
+if (!codeBlock || !unsupportedBlock) {
+  throw new Error("Code and Embed blocks need to be prepared before tests run");
+}
+
+test("Block page should contain key elements", async ({
+  page,
+  isMobile,
+  browserName,
+}) => {
+  await page.goto(codeBlock.blockPackagePath);
+
+  await expect(
+    page.locator(`h1:has-text("${codeBlock.displayName!}")`),
+  ).toBeVisible();
+
+  await expect(
+    page.locator(`p:has-text("${codeBlock.description!}")`),
+  ).toBeVisible();
+
+  await expect(page.locator(`text=@${codeBlock.author!}`)).toBeVisible();
+  await expect(page.locator(`text=@${codeBlock.author!}`)).toHaveAttribute(
+    "href",
+    `/@${codeBlock.author!}`,
+  );
+  await expect(page.locator(`text=V${codeBlock.version!}`)).toBeVisible();
+
+  const blockExample = codeBlock.examples?.[0] as Record<string, string>;
+
+  if (isMobile) {
+    await page.locator("text=Source Code").click();
+    await expect(page.locator("text=Data Source")).toBeVisible();
+  } else {
+    await expect(page.locator("text=Block Properties")).toBeVisible();
+  }
+  await expect(
+    page.locator("[data-testid='block-properties-tabpanel'] >> textarea"),
+  ).toHaveValue(JSON.stringify(blockExample, null, 2));
+
+  if (isMobile) {
+    await page.locator("text=Data Source").click();
+    await page.locator("text=Block Schema").click();
+  } else {
+    await page.locator("text=Block Schema").click();
+  }
+  await expect(
+    page.locator("[data-testid='block-schema-tabpanel']"),
+  ).toBeVisible();
+  await expect(
+    page.locator("[data-testid='block-schema-tabpanel']"),
+  ).not.toBeEmpty();
+
+  if (isMobile) {
+    await page.locator("text=Preview").click();
+  }
+
+  if (browserName !== "webkit") {
```

Comment:
Optional: add @todo & asana link

Reply:
Good catch. Added in [47592f2](https://github.com/blockprotocol/blockprotocol/pull/450/commits/47592f20cb448265c5baa3a94f80671da8609ae7)